internal/dumpfile/legacy_parse.go

```diff
@@ -0,0 +1,282 @@
+package dumpfile
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// readtrim returns a string read from the file or an error.
+func readtrim(rd *bufio.Reader) (string, error) {
+	l, err := rd.ReadString('\n')
+	if err == io.EOF && l == "" {
+		// We've run out of content.
+		return "", nil
+	} else if err != nil {
+		return "", err
+	}
+	l = strings.TrimRight(l, "\n")
+	return l, nil
+}
+
+var splittingregexp *regexp.Regexp
+
+func init() {
+	splittingregexp = regexp.MustCompile("[ \t]+")
+}
+
+// splitline splits the line based on a regexp and returns an array with not more than
+// count elements.
+func splitline(l string, count int) []string {
+	splits := splittingregexp.Split(strings.TrimLeft(l, "\t "), count)
+	// log.Printf("splitting %#v ➜ %#v", l, splits)
+	return splits
+}
+
+// loadhelper breaks out common load file parsing functionality for selected row
+// types.
+func loadhelper(rd *bufio.Reader, subl []string, fontname string, numcol, ndumped int, wintype WindowType) (Window, error) {
```

Comment:
I'm not sure why it's not returning a pointer to `Window` instead. It's a fairly large struct.

Reply:
I did this because of `dumpfile.Content.Windows`. We're copying the `dumpfile.Window` object regardless. But maybe `dumpfile.Content.Windows` should be `[]*Window`?

Reply:
My concern is that it's probably copying it multiple times because Go functions return values by copying to the stack. Consider this benchmark:
```
func newWindow() Window         { return Window{} }
func newWindowPointer() *Window { return &Window{} }

func BenchmarkNewWindow(b *testing.B) {
	for i := 0; i < b.N; i++ {
		w := newWindow()
		w.Column++
	}
}

func BenchmarkNewWindowPointer(b *testing.B) {
	for i := 0; i < b.N; i++ {
		w := newWindowPointer()
		w.Column++
	}
}
```
```
$ go test -bench=Window
goos: linux
goarch: amd64
pkg: github.com/rjkroege/edwood/internal/dumpfile
BenchmarkNewWindow-4          	97266338	        12.3 ns/op
BenchmarkNewWindowPointer-4   	1000000000	         0.292 ns/op
```

Reply:
Indeed. I had agreed with you. Anyway, I altered it.